core/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java

```diff
@@ -144,15 +144,15 @@ protected void innerToXContent(XContentBuilder builder, Params params) throws IO
             builder.endObject();
         }
         builder.endArray();
-        super.innerToXContent(builder, params);
     }
 
     @Override
-    protected void causeToXContent(XContentBuilder builder, Params params) throws IOException {
-        if (super.getCause() != null) {
-            // if the cause is null we inject a guessed root cause that will then be rendered twice so wi disable it manually
-            super.causeToXContent(builder, params);
+    protected boolean skipCauseInXContent(Params params) {
+        if (super.skipCauseInXContent(params)) {
+            return true;
         }
+        // if the cause is null we inject a guessed root cause that will then be rendered twice so we disable it manually
+        return super.getCause() == null;
```

Comment:
I don't follow the logic here and I wish we wouldn't need to have this skipCauseInXContent protected method. How does the behaviour here differ from the default one?